crates/snapshot/README.md

```diff
@@ -0,0 +1,85 @@
+
+Below are two diagrams illustrating the processes of creating static files (custom format: `NippyJar`) and querying them. A glossary is also provided to explain the different (linked) components involved in these processes.
+
+<details>
+  <summary>Creation diagram (<code>Snapshotter</code>)</summary>
+
+```mermaid
+graph TD;
+    I("BLOCK_HEIGHT % 500_000 == 0")--triggers-->SP(Snapshotter)
+    SP --> |triggers| SH["create_snapshot(block_range, SnapshotSegment::Headers)"]
+    SP --> |triggers| ST["create_snapshot(block_range, SnapshotSegment::Transactions)"]
+    SP --> |triggers| SR["create_snapshot(block_range, SnapshotSegment::Receipts)"]
+    SH --> CS["create_snapshot::&lt; T &gt;(DatabaseCursor)"]
+    ST --> CS
+    SR --> CS
+    CS --> |create| IF(NippyJar::InclusionFilters)
+    CS -- iterates --> DC(DatabaseCursor) -->HN{HasNext} 
+    HN --> |true| NJC(NippyJar::Compression)
+    NJC --store--> NJ
+    HN --> |false| NJ 
+    IF --store--> NJ(NippyJar)
+    NJ --freeze--> F(File)
+    F--"on success"--> SP1(Snapshotter)
+    SP1 --"sends BLOCK_HEIGHT"--> HST(HighestSnapshotTracker)
+    HST --"read by"-->Pruner
+    HST --"read by"-->DatabaseProvider
+    HST --"read by"-->SnapsotProvider
+    HST --"read by"-->ProviderFactory
+
+```
+</details>
+
+
+<details>
+  <summary>Query diagram (<code>Provider</code>)</summary>
+
+```mermaid
+graph TD;
+    RPC-->P
+    P("Provider::header(block_number)")-->PF(ProviderFactory)
+    PF--shares-->SP1("Arc(SnapshotProvider)")
+    SP1--shares-->PD(DatabaseProvider)
+    PF--creates-->PD
+    PD--check `HighestSnapshotTracker`-->PD
+    PD-->DC1{block_number <br> > <br> highest snapshot block}
+    DC1 --> |true| PD1("DatabaseProvider::header(block_number)")
+    DC1 --> |false| ASP("SnapshotProvider::header(block_number)")
+    PD1 --> MDBX
+    ASP --find correct jar and creates--> JP("SnapshotJarProvider::header(block_number)")
+    JP --"creates"-->SC(SnapshotCursor)
+    SC --".get_one&lt; HeaderMask&lt; Header  &gt; &gt;(number)"--->NJC("NippyJarCursor")
+    NJC--".row_by_number(row_index, mask)"-->NJ[NippyJar]
+    NJ--"&[u8]"-->NJC
+    NJC--"&[u8]"-->SC
+    SC--"Header"--> JP
+    JP--"Header"--> ASP
+```
+</details>
+
+
+### Glossary
+In descending order of abstraction hierarchy:
+
+[`Snapshotter`](../../crates/snapshot/src/snapshotter.rs#L20): A `reth` background service that **copies** data from the database to new snapshot files when the block height reaches a certain threshold (e.g., `500_000th`). Upon completion, it dispatches a notification about the higher snapshotted block to `HighestSnapshotTracker` channel. **It DOES NOT remove data from the database.**
+
+[`HighestSnapshotTracker`](../../crates/snapshot/src/snapshotter.rs#L22): A channel utilized by `Snapshotter` to announce the newest snapshot block to all components with a listener: `Pruner` (to know which additional tables can be pruned) and `DatabaseProvider`  (to know which data can be queried from the snapshots).
```

Comment:
I see, so the `HighestSnapshotTracker` channel acts both as a notification about changed snapshot tracker (for pruner) and to track the highest snapshot (for database)